src/acvm_interop/pwg.rs

```diff
@@ -82,25 +85,50 @@ impl PartialWitnessGenerator for Barretenberg {
                     .copied()
                     .chain(pub_key_y.to_vec())
                     .collect();
-                let pub_key: [u8; 64] = pub_key_bytes.try_into().unwrap();
+                let pub_key: [u8; 64] = pub_key_bytes.try_into().map_err(|v: Vec<u8>| {
+                    OpcodeResolutionError::BlackBoxFunctionFailed(
+                        func_call.name,
+                        format!("expected pubkey size {} but received {}", 64, v.len()),
+                    )
+                })?;
 
                 let mut signature = [0u8; 64];
                 for (i, sig) in signature.iter_mut().enumerate() {
-                    let _sig_i = inputs_iter.next().unwrap_or_else(|| {
-                        panic!("signature should be 64 bytes long, found only {i} bytes")
-                    });
+                    let _sig_i = inputs_iter.next().ok_or_else(|| {
+                        OpcodeResolutionError::BlackBoxFunctionFailed(
+                            func_call.name,
+                            format!("signature should be 64 bytes long, found only {i} bytes"),
+                        )
+                    })?;
                     let sig_i = witness_to_value(initial_witness, _sig_i.witness)?;
-                    *sig = *sig_i.to_be_bytes().last().unwrap()
+                    *sig = *sig_i.to_be_bytes().last().ok_or_else(|| {
+                        OpcodeResolutionError::BlackBoxFunctionFailed(
+                            func_call.name,
+                            "could not get last bytes".into(),
+                        )
```

Comment:
This error will never be thrown as it would require us to be using a 0 bit curve.

Reply:
We can also use a function such as https://github.com/noir-lang/acvm/pull/269/files#diff-c31b89a76cc86219fd7be804ffbb7dfe18d93b96924a0958f92a9b320572e2bbR6 here.